lytics.go

```diff
@@ -199,6 +205,54 @@ func (l *Client) Get(endpoint string, params url.Values, body io.Reader, respons
 	return nil
 }
 
+// Get prepares a post request and then executes using the Do method
+func (l *Client) Post(endpoint string, params url.Values, body interface{}, response, data interface{}) error {
+	method := "POST"
+
+	// get the formatted endpoint url
+	path, err := l.PrepUrl(endpoint, params, false)
+	if err != nil {
+		return err
+	}
+
+	payload, err := prepRequestBody(body)
+	if err != nil {
+		return err
+	}
+
+	// build the request
+	r, _ := http.NewRequest(method, path, payload)
+
+	// execute the request
+	err = l.Do(r, response, data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// prepBodyRequest takes the payload and returns an io.Reader
+func prepRequestBody(body interface{}) (io.Reader, error) {
+	// prep the body of the request
+	if body != nil {
```

Comment:
might consider something like this:

https://github.com/araddon/gou/blob/master/http.go#L53


also, in my code it is bad, the if data == nil doesn't need to be used, instead it can be a case in the switch statment.